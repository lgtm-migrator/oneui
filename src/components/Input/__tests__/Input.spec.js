import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Input from '../Input';

describe('<Input> that renders an input field', () => {
    const changeOptions = {
        target: {
            value: 'test',
        },
    };

    const onChangeMock = jest.fn();

    it('should render default input correctly', () => {
        const inputValue = 'Some value';
        const { container } = render(<Input value={inputValue} onChange={onChangeMock} />);
        expect(container.querySelector('input').value).toBe(inputValue);
        expect(container).toMatchSnapshot();
    });

    it('should add classes when props are changed', () => {
        const { container } = render(<Input context="bad" size="large" isBlock disabled />);
        expect(container.querySelector('input')).toHaveAttribute('disabled', '');
        expect(container).toMatchSnapshot();
    });

    // it('should call change callback correctly', () => {
    //     const { container, debug } = render(<Input onChange={onChangeMock} />);
    //     debug(container.querySelector('input'));
    //     fireEvent.change(container.querySelector('input'), changeOptions);
    //     expect(onChangeMock).toHaveBeenCalledWith(changeOptions);
    // });

    // it('should add string html attributes correctly', () => {
    //     const wrapper = shallow(<Input data-test="something" />);
    //     expect(wrapper.find('input').prop('data-test')).toEqual('something');
    // });
});
