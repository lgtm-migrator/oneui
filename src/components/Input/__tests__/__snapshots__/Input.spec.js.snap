// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> that renders an input field should add classes when props are changed 1`] = `
<div>
  <input
    class="Input Input--context_bad Input--isBlock Input--size_large"
    disabled=""
    type="text"
    value=""
  />
</div>
`;

exports[`<Input> that renders an input field should render default input correctly 1`] = `
<div>
  <input
    class="Input Input--context_brand"
    type="text"
    value="Some value"
  />
</div>
`;
